backend/src/workspace-documents/workspace-documents.service.ts

```diff
@@ -17,6 +17,42 @@ export class WorkspaceDocumentsService {
 		private configService: ConfigService
 	) {}
 
+	async updateTitle(
+		userId: string,
+		workspaceId: string,
+		documentId: string,
+		title: string
+	): Promise<void> {
+		try {
+			await this.prismaService.userWorkspace.findFirstOrThrow({
+				where: {
+					userId,
+					workspaceId,
+				},
+			});
+		} catch (e) {
+			throw new NotFoundException(
+				"The workspace does not exist, or the user lacks the appropriate permissions."
+			);
+		}
+
+		const document = await this.prismaService.document.findFirst({
+			where: {
+				id: documentId,
+				workspaceId: workspaceId,
+			},
+		});
+
+		if (!document) {
+			throw new NotFoundException("Document not found");
+		}
```

Comment:
We have already checked whether the document exists, so this is a redundant check.